### DISCUSSION TITLE
Double dot, potential issue?

### DISCUSSION THREAD

**Discussion by Czajnikowski** [pointfreeco/swift-case-paths#106] [General]
Hey guys,

I encountered a (strange?) behavior while working with that nice package.

Take a look at this code:
```swift
enum A {
    case b(B)
    case void
}

enum B {
    case void
    case some(C)
}

enum C {}

func x() {
    let cp1 = /A.void       // correct: CasePath<A, Void>
    let cp2 = /A.b..B.some  // correct: CasePath<A, C>
    let cp3 = /A.b..B.void  // build failed: Binary operator '..' cannot be applied to operands of type 'CasePath<A, B>' and 'B'
}
```

Shouldn't `cp3` be a valid `CasePath<A, Void>` as well 🤔?

**Comment by Czajnikowski** [comment]
I just figured out that I can work around it by wrapping the `B.void` in a closure (`let cp3 = /A.b..{ B.void }`)

**Reply by stephencelis** [reply]
Glad you worked it out!

One thing to note is that while we have embraced the `/` prefix operator, the other operators in the library have been exercised far less and we didn't think it was good to overload them more than they already are for the permutations we support.

For now we recommend using `.appending(path:)` for building up case paths. It's a little more verbose, but fights the language and compiler less:

```swift
(/A.b).appending(path: /B.void)
```

**Reply by Czajnikowski** [reply]
Alright, thanks for the hint!
